dotnet/src/webdriver/Firefox/Preferences.cs

```diff
@@ -152,39 +156,47 @@ private static bool IsWrappedAsString(string value)
 
         private bool IsSettablePreference(string preferenceName)
         {
-            return !this.immutablePreferences.ContainsKey(preferenceName);
+            return !this.immutablePreferences.Contains(preferenceName);
         }
 
-        private void SetPreferenceValue(string key, object value)
+        private void SetPreferenceValue(string key, JsonNode? value)
         {
             if (!this.IsSettablePreference(key))
             {
-                string message = string.Format(CultureInfo.InvariantCulture, "Preference {0} may not be overridden: frozen value={1}, requested value={2}", key, this.immutablePreferences[key], value.ToString());
+                string message = string.Format(CultureInfo.InvariantCulture, "Preference {0} may not be overridden: frozen value={1}, requested value={2}", key, this.preferences[key], value?.ToString());
```

Comment:
We can take advantage of the fact that `preferences` and `immutablePreferences` have the exact same values (hence this error message). We can store just the keys of the `immutablePreferences` and use a hashset instead,